Support llama3 training

dpalmasan opened the pull request:
Differential Revision: D70977130

* Add support to fine-tune llama
* Add a config example to fine-tune llama 3.2 1B

JacobSzwejbka commented on examples/llm_pte_finetuning/training_lib.py:
```diff
@@ -36,13 +37,24 @@ def __init__(
+        if not isinstance(logits, list):
+            labels = labels.reshape(-1)
+            logits = logits.reshape(-1, logits.size(-1))
         return self.loss(logits, labels)
```
should we return the prediction too?

silverguo committed:
Support llama3 training (#9149)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/9149

* Add support to fine-tune llama
* Add a config example to fine-tune llama 3.2 1B

Reviewed By: lucylq

Differential Revision: D70977130
